pkg/kind/cluster.go

```diff
@@ -100,13 +103,28 @@ func (c *Cluster) getConfig() ([]byte, error) {
 
 func NewCluster(name, kubeVersion, kubeConfigPath, kindConfigPath, extraPortsMapping string, cfg util.CorePackageTemplateConfig) (*Cluster, error) {
 	detectOpt, err := cluster.DetectNodeProvider()
+
 	if err != nil {
 		return nil, err
 	}
 	provider := cluster.NewProvider(detectOpt)
 
+	var rt runtime.IRuntime
+	if runtime.IsDockerAvailable() {
+		setupLog.Info("Detected runtim: Docker")
+		if rt, err = runtime.NewDockerRuntime(); err != nil {
+			return nil, err
+		}
+	} else if runtime.IsFinchAvailable() {
+		println("Detected runtime: Finch")
+		rt, _ = runtime.NewFinchRuntime()
+	} else {
+		return nil, errors.New("no runtime found")
+	}
+
```

Comment:
This bit should go to the runtime pkg. This kind of logic is already available in Kind. I think we should follow their logic: https://github.com/kubernetes-sigs/kind/blob/86c7c2e414662b42fcc36f028a604490bf5489a6/pkg/internal/runtime/runtime.go#L11

This way, users can use w/e runtime with the same configuration they would use for kind.

Reply:
sounds good. make the adjustments.

Reply:
I realized I didnt fully understand your comment above. will have to do another revisioin. will connect separately